File: 27 - Match Parentheses Pattern Validator/solution.py
def is_valid_parentheses(s):
    stack = []

    for char in s:
        if char == "(":
            stack.append(char)
        elif char == ")":
            if len(stack) == 0:
                return False
            stack.pop()
    

    return len(stack) == 0

# Test Cases
print(is_valid_parentheses("(abc)(de)f)g"))  # false

File: 27 - Match Parentheses Pattern Validator/solution.js
function isValidParentheses(str) {
  let stack = [];

  for (let char of str) {
    if (char === "(") {
      stack.push(char);
    } else if (char === ")") {
      if (stack.length === 0) {
        return false;
      } else {
        stack.pop();
      }
    }
  }

  return stack.length === 0;
}

// Test Cases
console.log(isValidParentheses("(abc)(de)f)g")); // false
